pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -881,9 +881,17 @@ public CompletableFuture<PendingAckHandle> pendingAckHandleFuture() {
     }
 
     @Override
-    public TransactionPendingAckStats getStats() {
+    public TransactionPendingAckStats getStats(boolean lowWaterMarks) {
         TransactionPendingAckStats transactionPendingAckStats = new TransactionPendingAckStats();
         transactionPendingAckStats.state = this.getState().name();
+        if (lowWaterMarks) {
+            transactionPendingAckStats.lowWaterMarks = this.lowWaterMarks;
+        }
+        if (individualAckOfTransaction != null) {
+            transactionPendingAckStats.ongoingTxnSize = individualAckOfTransaction.size();
```

Comment:
I am not sure about the synchronisation around "individualAckOfTransaction"
I suggest to declare a temporary variable and do not access twice the instance field

Reply:
There is no problem. IndividualAckOfTransaction will no longer become null after being initialized.